### ISSUE TITLE
rpm build faild

### ISSUE THREAD

**Issue by Jolin-blank** [netdata/netdata#10421]
```
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/charts2json.h:6,
                 from web/api/formatters/charts2json.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/json_wrapper.h:6,
                 from web/api/formatters/json_wrapper.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/rrdset2json.h:6,
                 from web/api/formatters/rrdset2json.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
make[2]: *** [web/api/formatters/json_wrapper.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [web/api/formatters/charts2json.o] Error 1
make[2]: *** [web/api/formatters/rrdset2json.o] Error 1
make[2]: Leaving directory `/root/rpmbuild/BUILD/netdata-v1.28.0'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/rpmbuild/BUILD/netdata-v1.28.0'
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.N2Mgf1 (%build)
```

**Comment by dgsus** [comment]
im getting the same but via the kickstart script on centos 6

no problem on dependencies
Loading /etc/centos-release ...
You should have EPEL enabled to install all the prerequisites.
Check: http://www.tecmint.com/how-to-enable-epel-repository-for-rhel-centos-6-5/
 > CentOS Version: 6.10 ...
 > Checking for epel ...
 > Detected CentOS 6.x ...
 > Checking for Okay ...

We detected these:
Distribution    : CentOS
Version         : 6.10
Codename        : Final
Package Manager : install_yum
Packages Tree   : centos
Detection Method: /etc/centos-release
Default Python v: 3

Searching for distro_sdk ...
Searching for autoconf_archive ...
WARNING
package autoconf-archive is not available in this system.
You may try to install without it.

Searching for autogen ...
WARNING
package autogen is not available in this system.
You may try to install without it.

Searching for libz_dev ...
 > Checking if package 'zlib-devel' is installed...
Searching for libuuid_dev ...
 > Checking if package 'libuuid-devel' is installed...
Searching for libmnl_dev ...
 > Checking if package 'libmnl-devel' is installed...
Searching for json_c_dev ...
 > Checking if package 'json-c-devel' is installed...
Searching for libuv ...
 > Checking if package 'libuv-devel' is installed...
Searching for lz4 ...
 > Checking if package 'lz4-devel' is installed...
Searching for openssl ...
 > Checking if package 'openssl-devel' is installed...
Searching for judy ...
Searching for libelf ...
 > Checking if package 'elfutils-libelf-devel' is installed...

All required packages are already installed. Now proceed to the next step.

...
...


make[2]: Leaving directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0/aclk'
Making all in spawn
make[2]: Entering directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0/spawn'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0/spawn'
make[2]: Entering directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0'
  CC     libnetdata/os.o
  CC     web/api/formatters/json_wrapper.o
  CC     web/api/formatters/charts2json.o
  CC     web/api/formatters/rrdset2json.o
  CC     exporting/read_config.o
  CC     exporting/clean_connectors.o
  CC     exporting/init_connectors.o
  CC     exporting/process_data.o
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/charts2json.h:6,
                 from web/api/formatters/charts2json.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/rrdset2json.h:6,
                 from web/api/formatters/rrdset2json.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from ./web/api/web_api_v1.h:6,
                 from web/api/formatters/rrd2json.h:6,
                 from web/api/formatters/json_wrapper.h:6,
                 from web/api/formatters/json_wrapper.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
make[2]: *** [web/api/formatters/charts2json.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [web/api/formatters/rrdset2json.o] Error 1
make[2]: *** [web/api/formatters/json_wrapper.o] Error 1
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from exporting/exporting_engine.h:6,
                 from exporting/read_config.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from exporting/exporting_engine.h:6,
                 from exporting/init_connectors.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from exporting/exporting_engine.h:6,
                 from exporting/clean_connectors.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
In file included from ./database/rrd.h:37,
                 from ./daemon/common.h:36,
                 from exporting/exporting_engine.h:6,
                 from exporting/process_data.c:3:
./database/../aclk/aclk_common.h:71: error: redefinition of typedef 'RRDHOST'
./database/rrd.h:7: note: previous declaration of 'RRDHOST' was here
make[2]: *** [exporting/read_config.o] Error 1
make[2]: *** [exporting/init_connectors.o] Error 1
make[2]: *** [exporting/clean_connectors.o] Error 1
make[2]: *** [exporting/process_data.o] Error 1
make[2]: Leaving directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/tmp/netdata-kickstart-h2laNYP9M0/netdata-v1.28.0'
make: *** [all] Error 2
 FAILED

 FAILED

 ABORTED  netdata-installer.sh exited with error

**Comment by underhood** [comment]
@Jolin-blank what is the os? How you build the RPM package?

**Comment by underhood** [comment]
@dgsus Thanks for info. I have made a PR that fixes this (#10470) and tested on CentOS 6 from archive.
Reason is older compiler on your CentOS 6 system.
Please take into account CentOS 6 is EOL and was removed also from CI/CD pipeline (which is the reason we didn't catch this problem).
As the CentOS 6 is EOL we cannot guarantee Netdata will work on it anymore and it is expected things can break in the future.
